### DISCUSSION TITLE
End Device sees data rate degradation when sending unconfirmed uplinks after gateway outage

### DISCUSSION THREAD

**Discussion by dmeehan1968** [Lora-net/LoRaMac-node#1071] [Q&A] [answered]
I've been doing some fault tolerance testing on my end devices, and one test has been to see what happens when the only gateway in view suffers an outage (e.g. loss of power or backhaul).

I have some devices sending debug data every 10 seconds at SF7, and these devices consistently start degrading data rate after the gateway outage occurs.  No other MLME or MCPS commands are being used, just unconfirmed uplinks.

It seems that the data rate degrades one level for each 32 uplinks, so DR5, DR4, DR3 etc.

Device ADR and Server ADR are enabled, but there is no ADR activity in the network server logs (Loriot) because there is no gateway in view that could be receiving the packets.  Network server shows no uplinks after gateway outage.

This seems contrary to my understanding of how LoRa works.  Why would the device be aware of gateway outage when only sending unconfirmed uplinks?

**Comment by mluis1** [comment, marked as answer]
The described behavior is according to LoRaWAN specification.

Please read LoRaWAN 1.0.4 specification chapter "4.3.1.1 Adaptive data-rate control in frame header (ADR, ADRACKReq in FCtrl)"
Please also note that this behavior exists since LoRaWAN 1.0.0 specification.

At the end of the chapter you will see an example table describing the expected behavior.

![image](https://user-images.githubusercontent.com/5763295/112484056-e5bf2280-8d79-11eb-9b19-3c67035176c3.png)

After `ADR_ACK_LIMIT` up links the end-device sets the `ADRACKReq` bit on frame `FCtrl` field.

By setting the `ADRACKReq` bit to 1 the end-device is requesting the network server to execute a down link.
In case a down link is received the `ADRACKCnt` is reset to 0 and the process re-starts.
In case a down link is not received the end-device will lower its data rate every `ADR_ACK_DELAY`. Once lowest data rate is reached the end-device enables default channels (Dynamic Channel plans) or enables all channels (Fixed Channel plans).

**Comment by nestorayuso** [comment]
This is the expected behaviour. 
Even if node is using unconfirmed messages at Application Level, the LoRaWAN protocol need bidirectional communication at MAC Level.

At least one message reception every 32 messages sent, if not, the datarate is lowered.
